### PR TITLE
User Can Send Message from Inbox

### PR DESCRIPTION
**Pull request by sethnejame**
Adding feature test: that a signed in User has an inbox that can send messages .  See user story below:

  As a user,
    In order to communicate with another user
    I should have an inbox that sends messages

### CODE REVIEW

File: features/inbox.feature [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    Given the following user exists
+        | email              |  password  | name       |
+        | hackerman@gmail.com|  hackerman |  Hackerman |
+    Given I visit the site
+    Then I should see "Login"
+    When I click "Login"
+    And I enter "hackerman@gmail.com" into the "Email" field
+    And I enter "hackerman" into the "Password" field
+    And I click "Log in"
+    Then I should see "Signed in successfully."
```
Comment (oliverochman) [inline features/inbox.feature]: Remove this assertion from the background

Comment (oliverochman) [reply features/inbox.feature]: we never write assertions in the background

Comment (oliverochman) [reply features/inbox.feature]: plus we don't test the login flow here so it is unnecessary

File: features/inbox.feature [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    
+
+  Scenario: View Inbox when User Logs In
```
Comment (oliverochman) [inline features/inbox.feature]: Is this scenario needed when we know that a user can compose a message?

File: features/inbox.feature [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    As a user,
+    In order to read & send messages
+    I should have an inbox that receives and sends messages
+
+Background: 
+    Given the following user exists
+        | email              |  password  | name       |
+        | hackerman@gmail.com|  hackerman |  Hackerman |
+    Given I visit the site
+    Then I should see "Login"
```
Comment (oliverochman) [inline features/inbox.feature]: If the user can click it, they can see it
Please remove this step

File: features/inbox.feature [hunk 4]
```diff
@@ -0,0 +1,34 @@
+Feature: Have a working Inbox
+    As a user,
+    In order to read & send messages
+    I should have an inbox that receives and sends messages
+
+Background: 
+    Given the following user exists
+        | email              |  password  | name       |
+        | hackerman@gmail.com|  hackerman |  Hackerman |
+    Given I visit the site
```
Comment (oliverochman) [inline features/inbox.feature]: Change to `And I visit the site`

File: features/inbox.feature [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+    
+
+  Scenario: View Inbox when User Logs In
+    When I click on "Inbox" button
+    Then I should see a "Compose" button and "Inbox" and "Sent" and "Trash"
+    
+
+  Scenario: Send Messages to another user
+    When I click on "Inbox" button
+    Then I should see a "Compose" button and "Inbox" and "Sent" and "Trash"
```
Comment (oliverochman) [inline features/inbox.feature]: This is out of scope for this feature, you dont need to test for "Sent" & "Trash" when you test the composing of messages

File: features/inbox.feature [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+    When I click on "Compose" in my Inbox
+    Then I should see "Recipients" and "Subject" and "Message"
```
Comment (oliverochman) [inline features/inbox.feature]: Same here

File: features/inbox.feature [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+    And I select a user from the "Recipient" box
```
Comment (oliverochman) [inline features/inbox.feature]: What user is this?

File: features/step_definitions/inbox_steps.rb [hunk 8]
```diff
@@ -0,0 +1,42 @@
+Given("the following user exists") do |table|
+  table.hashes.each do |table|
```
Comment (oliverochman) [inline features/step_definitions/inbox_steps.rb]: change `|table|` to `|user|`

File: features/step_definitions/inbox_steps.rb [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+    FactoryBot.create(:user, table)
+  end
+end
+
+Given("I visit the site") do
+  visit(root_path)
```
Comment (oliverochman) [inline features/step_definitions/inbox_steps.rb]: please stay consistent with poetry mode
`visit root_path`

File: features/step_definitions/inbox_steps.rb [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+
+When("I click {string}") do |button|
+  click_on button
+end
+
+When("I enter {string} into the {string} field") do |input, field|
+  fill_in field, with: input
+end
+
+Then("I click on {string} button") do |button|
```
Comment (oliverochman) [inline features/step_definitions/inbox_steps.rb]: Why do you need this one when you have the step definition on line 15?

File: features/step_definitions/inbox_steps.rb [hunk 11]
```diff
@@ -0,0 +1,42 @@
...
+When("I click on {string} in my Inbox") do |button|
+  click_on button
+end
+
+Then("I should see {string} and {string} and {string}") do |string, string2, string3|
+  expect(page).to have_content string
+end
+
+Then("I select a user from the {string} box") do |string|
+  select 'Hackerman', from: 'conversation[recipients][]'
```
Comment (oliverochman) [inline features/step_definitions/inbox_steps.rb]: this is hardcoded

File: features/inbox.feature [hunk 12]
```diff
@@ -0,0 +1,34 @@
+Feature: Have a working Inbox
+    As a user,
+    In order to read & send messages
+    I should have an inbox that receives and sends messages
+
+Background: 
+    Given the following user exists
+        | email              |  password  | name       |
+        | hackerman@gmail.com|  hackerman |  Hackerman |
```
Comment (oliverochman) [inline features/inbox.feature]: you need to have a user to send the message to

File: features/inbox.feature [hunk 13]
```diff
@@ -0,0 +1,34 @@
+Feature: Have a working Inbox
```
Comment (oliverochman) [inline features/inbox.feature]: I would change this feature to only handle the composing of a message

File: features/inbox.feature (hunk 1 above)
Comment (sethnejame) [reply features/inbox.feature]: Removed!

File: features/inbox.feature (hunk 3 above)
Comment (sethnejame) [reply features/inbox.feature]: Removed!

File: features/inbox.feature (hunk 4 above)
Comment (sethnejame) [reply features/inbox.feature]: Changed!

File: features/inbox.feature (hunk 5 above)
Comment (sethnejame) [reply features/inbox.feature]: Fixed!

File: features/inbox.feature (hunk 2 above)
Comment (sethnejame) [reply features/inbox.feature]: Removed

File: features/step_definitions/inbox_steps.rb (hunk 9 above)
Comment (sethnejame) [reply features/step_definitions/inbox_steps.rb]: Done

File: features/step_definitions/inbox_steps.rb (hunk 10 above)
Comment (sethnejame) [reply features/step_definitions/inbox_steps.rb]: Refactored

File: features/step_definitions/inbox_steps.rb (hunk 11 above)
Comment (sethnejame) [reply features/step_definitions/inbox_steps.rb]: Copy that. . .we have created another user as an example user to send to called "Lonely Gal" and she is hardcoded

File: features/inbox.feature (hunk 12 above)
Comment (sethnejame) [reply features/inbox.feature]: Done

File: features/inbox.feature (hunk 13 above)
Comment (sethnejame) [reply features/inbox.feature]: Done

File: features/step_definitions/inbox_steps.rb [hunk 14]
```diff
@@ -0,0 +1,35 @@
...
+end
+
+Given("I visit the site") do
+  visit root_path
+end
+
+Given(/^I am logged in as "([^"]*)"$/) do |name|
+      user = User.find_by(name: name)
+      login_as(user, scope: :user)
+    end
```
Comment (oliverochman) [inline features/step_definitions/inbox_steps.rb]: fix indentation

File: features/step_definitions/inbox_steps.rb [hunk 15]
```diff
@@ -0,0 +1,35 @@
...
+
+Then("I should see {string}") do |content|
```
Comment (oliverochman) [inline features/step_definitions/inbox_steps.rb]: This is an assertion step
Please move it to assertion_steps.rb

File: features/step_definitions/inbox_steps.rb [hunk 16]
```diff
@@ -0,0 +1,35 @@
...
+
+When("I click {string}") do |button|
+  click_on button
+end
+
+When("I enter {string} into the {string} field") do |input, field|
+  fill_in field, with: input
+end
+
+Then("I should see {string} and {string} and {string}") do |string, string2, string3|
```
Comment (oliverochman) [inline features/step_definitions/inbox_steps.rb]: This is an assertion step
Please move it to assertion_steps.rb

File: features/inbox.feature [hunk 17]
```diff
@@ -0,0 +1,24 @@
...
+    Given the following user exists
+        | email              |  password  | name       |
+        | hackerman@gmail.com|  hackerman |  Hackerman |
+        | lonelygal@gmail.com|  lonelygal |  LonelyGal |
+    Given I am logged in as "Hackerman"
+    And I visit the site
+
+  Scenario: Send Messages to another user
+    When I click "Inbox"
+    Then I should see "Compose"
```
Comment (oliverochman) [inline features/inbox.feature]: If you can click it, you can see it
Please remove this line

File: features/inbox.feature (hunk 6 above)
Comment (oliverochman) [reply features/inbox.feature]: Please remove this, not necessary to test this

File: features/step_definitions/inbox_steps.rb [hunk 18]
```diff
@@ -0,0 +1,35 @@
...
+  expect(page).to have_content string
+end
+
+Then("I select a user from the {string} box") do |string|
```
Comment (oliverochman) [inline features/step_definitions/inbox_steps.rb]: Please change to `When("I select a user from the {string} box") do |string|`

Comment (oliverochman) [reply features/step_definitions/inbox_steps.rb]:
```
When("I select {string} named {string}") do |field, recipient|
  select recipient, from: field
end
```
`And I select "Recipients" named "LonelyGal"`

File: features/step_definitions/inbox_steps.rb (hunk 14 above)
Comment (oliverochman) [reply features/step_definitions/inbox_steps.rb]: can you try writing line 13 in poetry mode

Comment (sethnejame) [reply features/step_definitions/inbox_steps.rb]: Done

File: features/step_definitions/inbox_steps.rb (hunk 15 above)
Comment (sethnejame) [reply features/step_definitions/inbox_steps.rb]: Done

File: features/step_definitions/inbox_steps.rb (hunk 16 above)
Comment (sethnejame) [reply features/step_definitions/inbox_steps.rb]: Done

File: features/inbox.feature (hunk 17 above)
Comment (sethnejame) [reply features/inbox.feature]: Done

File: features/step_definitions/inbox_steps.rb (hunk 18 above)
Comment (sethnejame) [reply features/step_definitions/inbox_steps.rb]: Done

File: features/inbox.feature (hunk 6 above)
Comment (sethnejame) [reply features/inbox.feature]: Done

File: features/inbox.feature (hunk 7 above)
Comment (sethnejame) [reply features/inbox.feature]: Done

File: features/step_definitions/inbox_steps.rb (hunk 8 above)
Comment (sethnejame) [reply features/step_definitions/inbox_steps.rb]: Done

File: spec/factories/users.rb [hunk 19]
```diff
@@ -1,5 +1,8 @@
 FactoryBot.define do
   factory :user do
+    # name { "MyString" }
+    # email { "MyString" }
+    # password { "MyString" }
```
Comment (faraznaeem) [inline spec/factories/users.rb]: Please remove comments

Comment (sethnejame) [reply spec/factories/users.rb]: Thy will be done!
